### PR TITLE
Fixes #18381: outcome should be enums rather than hardcoded

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18381

### CODE REVIEW

**Comment by gpoblon** [comment]
Commit modified

File: rudder-lang/src/technique/outcome.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2019-2020 Normation SAS
+
+use crate::error::*;
+use std::{fmt, str::FromStr};
+use strum::{EnumIter, IntoEnumIterator};
+
+#[derive(EnumIter, PartialEq, Debug)]
+pub enum ConditionOutcome {
```
Comment (peckpeck) [inline rudder-lang/src/technique/outcome.rs:9]: Literal outcomes as used in return statements should only be Repaired, Kept or Error

**Comment by gpoblon** [comment]
Commit modified

**Comment by peckpeck** [comment]
I see a commit modified message, but no change, could you do fixups during review so that I can easily see dif of changes

**Comment by gpoblon** [comment]
yes I just rebased the branch and started looking into a fix but I'd need to discuss about it with you

**Comment by gpoblon** [comment]
yes I just rebased the branch and started looking into a fix but I'd need to discuss about it with you

**Pull request closed without merging**